tools/resource-gen.go

```diff
@@ -218,48 +267,56 @@ func OnKumaResourceMessage(f ProtoMessageFunc) ProtoMessageFunc {
 }
 
 func main() {
-	flag.StringVar(&Package, "package", "mesh", "Package to generate")
+	var gen string
+	flag.StringVar(&gen, "generator", "", "the type of generator to run options: (type,ws)")
 	flag.Parse()
 
-	fmt.Fprintf(Output, `
-// Generated by tools/resource-gen.
-// Run "make generate" to update this file.
-
-// nolint:whitespace
-package %s
-
-import (
-	"fmt"
-
-	mesh_proto "github.com/kumahq/kuma/api/%s/v1alpha1"
-	"github.com/kumahq/kuma/pkg/core/resources/model"
-	"github.com/kumahq/kuma/pkg/core/resources/registry"
-)
-
-`, Package, Package)
-
-	var types []protoreflect.MessageType
-
-	protoregistry.GlobalTypes.RangeMessages(
-		OnKumaResourceMessage(func(m protoreflect.MessageType) bool {
-			types = append(types, m)
-			return true
-		}))
-
-	// Sort by name so the output is deterministic.
-	sort.Slice(types, func(i, j int) bool {
-		return types[i].Descriptor().FullName() < types[j].Descriptor().FullName()
-	})
-
-	for _, t := range types {
-		GenerateResource(t)
-	}
-
-	out, err := format.Source(Output.Bytes())
-	if err != nil {
-		fmt.Fprintf(os.Stderr, "%s\n", err)
-		os.Exit(1)
+	for _, pkg := range []string{"system", "mesh"} {
+		var types []protoreflect.MessageType
+		protoregistry.GlobalTypes.RangeMessages(
+			OnKumaResourceMessage(pkg, func(m protoreflect.MessageType) bool {
+				types = append(types, m)
+				return true
+			}))
+
+		// Sort by name so the output is deterministic.
+		sort.Slice(types, func(i, j int) bool {
+			return types[i].Descriptor().FullName() < types[j].Descriptor().FullName()
+		})
+
+		var resources []ResourceInfo
+		apisPath := fmt.Sprintf("pkg/core/resources/apis/%s", pkg)
+		for _, t := range types {
+			resourceInfo := ToResourceInfo(t)
+			resources = append(resources, resourceInfo)
+		}
+		globalTemplates := map[string]*template.Template{}
+		switch gen {
+		case "type":
+			globalTemplates[path.Join(apisPath, "generated_resources.go")] = ResourceTemplate
+		case "ws":
+			globalTemplates[path.Join("pkg/api-server/definitions", fmt.Sprintf("generated_%s_definitions.go", pkg))] = WsResourceTemplate
+		default:
+			panic(fmt.Sprintf("%s is not a valid generator option\n", gen))
+		}
```

Comment:
nit: if you add a newline between related blocks, it separates the code into paragraphs, which makes it easier to read. This might be my own idiosyncrasy though :)